internal/connector/pingone_platform/resources/pingone_trusted_email_address.go

```diff
@@ -0,0 +1,79 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/pingidentity/pingctl/internal/connector"
+	"github.com/pingidentity/pingctl/internal/logger"
+)
+
+// Verify that the resource satisfies the exportable resource interface
+var (
+	_ connector.ExportableResource = &PingoneTrustedEmailAddressResource{}
+)
+
+type PingoneTrustedEmailAddressResource struct {
+	clientInfo *connector.SDKClientInfo
+}
+
+// Utility method for creating a PingoneTrustedEmailAddressResource
+func TrustedEmailAddress(clientInfo *connector.SDKClientInfo) *PingoneTrustedEmailAddressResource {
+	return &PingoneTrustedEmailAddressResource{
+		clientInfo: clientInfo,
+	}
+}
+
+func (r *PingoneTrustedEmailAddressResource) ExportAll() (*[]connector.ImportBlock, error) {
+	l := logger.Get()
+
+	l.Debug().Msgf("Fetching all %s resources...", r.ResourceType())
+
+	apiExecuteFunc := r.clientInfo.ApiClient.ManagementAPIClient.TrustedEmailDomainsApi.ReadAllTrustedEmailDomains(r.clientInfo.Context, r.clientInfo.ExportEnvironmentID).Execute
+	apiFunctionName := "ReadAllTrustedEmailDomains"
+
+	emailDomainEmbedded, err := GetManagementEmbedded(apiExecuteFunc, apiFunctionName, r.ResourceType())
+	if err != nil {
+		return nil, err
+	}
+
+	importBlocks := []connector.ImportBlock{}
+
+	l.Debug().Msgf("Generating Import Blocks for all %s resources...", r.ResourceType())
+
+	for _, trustedEmailDomain := range emailDomainEmbedded.GetEmailDomains() {
+		trustedEmailDomainId, trustedEmailDomainIdOk := trustedEmailDomain.GetIdOk()
+		trustedEmailDomainName, trustedEmailDomainNameOk := trustedEmailDomain.GetDomainNameOk()
+
+		if trustedEmailDomainIdOk && trustedEmailDomainNameOk {
+			apiExecuteFunc := r.clientInfo.ApiClient.ManagementAPIClient.TrustedEmailAddressesApi.ReadAllTrustedEmailAddresses(r.clientInfo.Context, r.clientInfo.ExportEnvironmentID, *trustedEmailDomainId).Execute
+			apiFunctionName := "ReadAllTrustedEmailAddresses"
+
+			trustedEmailAddressEmbedded, err := GetManagementEmbedded(apiExecuteFunc, apiFunctionName, r.ResourceType())
+			if err != nil {
+				return nil, err
+			}
+
+			for _, trustedEmailAddress := range trustedEmailAddressEmbedded.GetTrustedEmails() {
+				if trustedEmailAddress.EmailAddress != "" {
+					trustedEmailAddressId, trustedEmailAddressIdOk := trustedEmailAddress.GetIdOk()
+
+					trustedEmailAddressDomainId, trustedEmailAddressDomainIdOk := trustedEmailAddress.GetDomainIdOk()
+
+					if trustedEmailAddressIdOk && trustedEmailAddressDomainIdOk {
+						importBlocks = append(importBlocks, connector.ImportBlock{
+							ResourceType: r.ResourceType(),
+							ResourceName: fmt.Sprintf("%s/%s", *trustedEmailDomainName, trustedEmailAddress.EmailAddress),
```

Comment:
Do we normally put slashes in resource names?

Reply:
corrected